SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialPlayerSpawner.cpp

```diff
@@ -198,40 +198,58 @@ void USpatialPlayerSpawner::ReceivePlayerSpawnRequestOnServer(const Worker_Comma
+	// Else if a PlayerStart can't be found THEN we could be on the wrong worker type, so forward to the GameMode authoritative server.
+	// 
+	// This implementation depends on:
+	// 1) the load-balancing strategy having the same rules for PlayerStart Actors and Characters / Controllers / Player States or,
+	// 2) the authoritative virtual worker ID for a PlayerStart Actor not changing during the lifetime of a deployment.
+	check (NetDriver->LoadBalanceStrategy != nullptr)
+
+	// We need to specifically extract the URL from the PlayerSpawn request for finding a PlayerStart.
+	const FURL Url = PlayerSpawner::ExtractUrlFromPlayerSpawnParams(SpawnPlayerRequest);
+
+	// Find a PlayerStart Actor on this server.
+	AActor* PlayerStartActor = NetDriver->GetWorld()->GetAuthGameMode()->FindPlayerStart(nullptr, Url.Portal);
+
+	// If the PlayerStart is authoritative locally, spawn the player locally.
+	if (PlayerStartActor != nullptr && NetDriver->LoadBalanceStrategy->ShouldHaveAuthority(*PlayerStartActor))
 	{
-		// We need to specifically extract the URL from the PlayerSpawn request for finding a PlayerStart.
-		const FURL Url = PlayerSpawner::ExtractUrlFromPlayerSpawnParams(SpawnPlayerRequest);
-		AActor* PlayerStartActor = NetDriver->GetWorld()->GetAuthGameMode()->FindPlayerStart(nullptr, Url.Portal);
+		UE_LOG(LogSpatialPlayerSpawner, Verbose, TEXT("Handling SpawnPlayerRequest request locally. Client worker ID: %s."), *ClientWorkerId);
+		PassSpawnRequestToNetDriver(SpawnPlayerRequest, PlayerStartActor);
+		return;
+	}
 
-		if (!NetDriver->LoadBalanceStrategy->ShouldHaveAuthority(*PlayerStartActor))
-		{
-			// If we fail to forward the spawn request, we default to the normal player spawning flow.
-			const bool bSuccessfullyForwardedRequest = ForwardSpawnRequestToStrategizedServer(SpawnPlayerRequest, PlayerStartActor, ClientWorkerId);
-			if (bSuccessfullyForwardedRequest)
-			{
-				return;
-			}
-		}
-		else
-		{
-			UE_LOG(LogSpatialPlayerSpawner, Verbose, TEXT("Handling SpawnPlayerRequest request locally. Client worker ID: %s."), *ClientWorkerId);
-			PassSpawnRequestToNetDriver(SpawnPlayerRequest, PlayerStartActor);
-			return;
-		}
+	VirtualWorkerId VirtualWorkerToForwardTo = SpatialConstants::INVALID_VIRTUAL_WORKER_ID;
+
+	// If we can't find a PlayerStart Actor, the PlayerSpawner authoritative worker may be part of a layer
+	// which has a limited view of the world and/or shouldn't be processing player spawning. In this case,
+	// we attempt to forward to the worker authoritative over the GameMode, as we assume the FindPlayerStart
+	// implementation may depend on authoritative game mode logic. We pass a null object ref so that the
+	// forwarded worker knows to search for a PlayerStart.
+	if (PlayerStartActor == nullptr)
+	{
+		VirtualWorkerToForwardTo = NetDriver->LoadBalanceStrategy->WhoShouldHaveAuthority(*UGameplayStatics::GetGameMode(GetWorld()));
```

Comment:
Is it worth logging/checking that `VirtualWorkerToForwardTo` is not this worker? In case for whatever reason FindPlayerStart returned nullptr but is authoritive for spawning

Reply:
Yep nice suggestion